object/asset_status.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 The casbin Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package object
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/casvisor/casvisor/metric"
+	"xorm.io/core"
+)
+
+func RunCheckAssetStatus() error {
+	assets := []*Asset{}
+	err := adapter.Engine.Where("category = ?", "Machine").Find(&assets)
+	if err != nil {
+		return err
+	}
+
+	runner := metric.Runner{}
+	for _, asset := range assets {
+		// use a copy of asset to avoid data race
+		a := asset
+		runner.Add(func() error {
+			return CheckAssetStatus(a)
+		})
+	}
+	runner.Wait()
+	return nil
+}
+
+func CheckAssetStatus(asset *Asset) error {
+	err := ICMPPingHost(asset.Endpoint)
+	if err != nil {
+		asset.Status = AssetStatusStopped
+		asset.SshStatus = AssetStatusStopped
+		_, _ = adapter.Engine.ID(core.PK{asset.Owner, asset.Name}).Cols("status", "ssh_status").Update(asset)
```

Comment:
So many DB update sentences, can you do only one DB update at the end of this function?